### PR TITLE
Add tests for _npe2.py

### PR DESCRIPTION
**Pull request by ppwadhwa**
# Description

Adding tests for _npe2.py file.  Let me know if you have any thoughts or suggestions!  

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by ppwadhwa** [comment]
I see that the tests aren't running because I imported something from qtpy.  I did this because I wanted to test the function `populate_qmenu` by adding to a QMenu.  I tried to do this at first by creating a viewer with `make_napari_viewer`, but this viewer already included the npe2 plugins in the menu.  So, it doesn't add them with `populate_qmenu`.  Do we need this function?  Is there a better way to write a test for this?

**Comment by ppwadhwa** [comment]
OK.  well, I guess I need to dive in a bit more here.  These tests pass on my machine, but I must be missing something to get them to pass here...

**Comment by sofroniewn** [comment]
I'm not sure why tests are failing - maybe you can merge in main and try rerunning. Also maybe @nclack or @tlambert03 have an idea here

**Comment by tlambert03** [comment]
your `uses_sample_plugin` plugin fixture isn't working.  Not sure why it's working for you locally... but there's no sample plugin in this PR (at `plugins/_tests/sample/my_plugin/napari.yaml`)

**Comment by ppwadhwa** [comment]
~~@tlambert03 oh, you're right.  I did take the sample plugin from npe2, so maybe because that's installed in my environment is why its somehow working?  I am not sure.  but, I'm looking into it.  I'll probably bring it over.~~

**Comment by ppwadhwa** [comment]
nope, I had the sample plugin in my repo.  I just missed it on the git add---just as you said @sofroniewn yesterday.

**Comment by ppwadhwa** [comment]
The tests are failing because of this file: `napari/layers/utils/_tests/test_style_encoding.py`. I merged from main.  I am not sure what I may be missing that is making these tests fail.  Anyone have some ideas?

**Comment by tlambert03** [comment]
See #4150

**Comment by sofroniewn** [comment]
#4150 is now merged - can you update from main here @ppwadhwa and let's see if all tests then pass - thanks

**Review by tlambert03 — COMMENTED** [review]
my first thought here is that I don't love having to copy the entire `sample/my_plugin` plugin code over from npe2.  maybe we should expose a sample plugin fixture from npe2 as a pytest [plugin](https://docs.pytest.org/en/6.2.x/writing_plugins.html)

**Comment by ppwadhwa** [comment]
@tlambert03 ok.  that makes sense.  I will read the link you posted and try to get that in place.

File: napari/plugins/_npe2.py [hunk 1]
```diff
@@ -131,9 +131,13 @@ def populate_qmenu(menu: QMenu, menu_key: str):
...
-            cmd = pm.get_command(item.command)
-            action = menu.addAction(cmd.title)
-            action.triggered.connect(lambda *args: cmd.exec(args=args))
+            try:
+                cmd = pm.get_command(item.command)
+                action = menu.addAction(cmd.title)
+                action.triggered.connect(lambda *args: cmd.exec(args=args))
+            except KeyError:
+                # this does not have a command associated with it.
+                pass
```
Comment (ppwadhwa) [inline napari/plugins/_npe2.py]: I think I had a problem here when I was writing tests using the sample plugin fixture.  There was not a command associated with the item.  I don't know if this is something that should be allowed.   If you are going to put it in a menu, should it have a command?  Should we put a warning here if one is not found?

Comment (tlambert03) [reply napari/plugins/_npe2.py]: see the schema [here](https://github.com/napari/npe2/blob/7791e1016641a6193e91e06b418db5273a678652/npe2/manifest/menus.py#L45-L50).  All menu items that are not submenus must have a command

**Comment by ppwadhwa** [comment]
There are some tests failing: 1 in test_layers_list.py and a few in test_layer_save.py, which are not related to this PR.  I haven't looked into it, but is anyone yet?

**Comment by ppwadhwa** [comment]
Just FYI--I'm taking one more pass through these tests to make sure they make sense and will add docstrings to explain what I'm trying to test.

**Comment by tlambert03** [comment]
@ppwadhwa did you see my pr to your branch?

**Comment by ppwadhwa** [comment]
@tlambert03 I have not seen it and maybe I'm just slow today, but I can't find it.  did I mess something up?  Can you link it here?

**Comment by tlambert03** [comment]
https://github.com/ppwadhwa/napari/pull/3

**Comment by ppwadhwa** [comment]
ok, I found it.  nevermind!

**Review by tlambert03 — APPROVED** [review]

**Pull request merged**